src/main/java/Duke.java

```diff
@@ -1,10 +1,136 @@
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Scanner;
+import java.util.Set;
+import task.DeadlineTask;
+import task.EventTask;
+import task.Task;
+
 public class Duke {
+
+    /**
+     * User commands
+     */
+    private static final String COMMAND_EXIT = "bye";
+    private static final String COMMAND_LIST = "list";
+    private static final String COMMAND_MARK = "mark";
+    private static final String COMMAND_UNMARK = "unmark";
+    private static final Set<String> COMMANDS_TASK = new HashSet<>(Arrays.asList("todo", "deadline", "event"));
+
+    private static List<Task> userTasks = new ArrayList<>();
+
+    private static boolean isRunning;
+    private static Scanner scanner;
+
+    // Overloaded default for printDivider
+    private static void printDivider() {
+        printDivider(60);
+    }
+
+    private static void printDivider(int width) {
+        System.out.println("_".repeat(width));
+    }
+
+    private static void printUserTasks() {
+        for (int index = 1; index <= userTasks.size(); index++) {
+            System.out.println(index + ". " + userTasks.get(index - 1));
+        }
+    }
+
+    private static void setUserTaskState(int userIndex, boolean isDone) {
+        // Tasks are 0-indexed, user index is 1-indexed
+        int index = userIndex - 1;
+        if (index < 0 || index >= userTasks.size()) {
+            System.out.println("Oops, not quite sure what task you're referring to...");
+            return;
+        }
+
+        userTasks.get(index).setTaskState(isDone);
+        if (isDone) {
+            System.out.println("Nice! I've marked this task as done");
+            System.out.println(userTasks.get(index));
+        } else {
+            System.out.println("Ok, I've marked this task as not done yet:");
+            System.out.println(userTasks.get(index));
+        }
+    }
+
+    private static void handleAddUserTask(String cmd, String[] splitInput) {
+        Task task = null;
+
+        if (cmd.equals("todo")) {
+            // Syntax: todo <desc>
+            task = new Task(splitInput[1]);
+
+        } else if (cmd.equals("deadline")) {
+            // Syntax: deadline <desc> /by <deadline>
+            String[] params = splitInput[1].split("/by");
```

Comment:
I like how you named multi-valued variables using a plural term as it allows readers to distinguish clearly between single-valued and multi-valued variables.